pkg/cli/admin/ocpcertificates/regeneratemco/command.go

```diff
@@ -0,0 +1,297 @@
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+const (
+	OneYear    = 365 * 24 * time.Hour
+	caExpiry   = 10 * OneYear
+	caRefresh  = 9 * OneYear
+	keyExpiry  = caExpiry
+	keyRefresh = caRefresh
+
+	mcoNamespace   = "openshift-machine-config-operator"
+	mapiNamespace  = "openshift-machine-api"
+	controllerName = "OcMachineConfigServerRotator"
+	mcsName        = "machine-config-server"
+
+	// mcsTlsSecretName is created by the installer and is not owned by default
+	mcsTlsSecretName = mcsName + "-tls"
+	// signerName is the name injected into the certificate as the signer
+	signerName = "oc-for-machine-config-server"
+	// openshiftOrg is the openshift organizational unit
+	openshiftOrg = "openshift"
+
+	// legacyRootCANamespace is the namespace of the legacy root CA
+	legacyRootCANamespace = "kube-system"
+	// legacyRootCA is the name of the configmap holding the MCS CA created by openshift-install
+	legacyRootCA = "root-ca"
+
+	generatedAnnotationKey   = "openshift.io/generated-by"
+	generatedAnnotationValue = "oc"
+
+	// newMCSCASecret is the location of the CA after rotation
+	newMCSCASecret   = "machine-config-server-ca"
+	CASecretCertName = "tls.crt"
+	userDataKey      = "userData"
+
+	// mcoManagedWorkerSecret is the unused, MCO-managed stub ignition for workers
+	mcoManagedWorkerSecret = "worker-user-data-managed"
+	// mcoManagedMasterSecret is the unused, MCO-managed stub ignition for masters
+	mcoManagedMasterSecret = "master-user-data-managed"
+	// mcsLabelSelector is used to select the MCS pods
+	mcsLabelSelector = "k8s-app=machine-config-server"
+)
+
+var (
+	regenerateMCOLong = templates.LongDesc(`
+		Regenerate the Machine Config Operator certificates for an OCP v4 cluster.
+
+		More information about these certificates are in the product documentation;
+		visit [docs.openshift.com](https://docs.openshift.com)
+		and select the version of OpenShift you are using.
+
+		Experimental: This command is under active development and may change without notice.
+	`)
+
+	regenerateMCOExample = templates.Examples(`
+		oc adm certificates regenerate-mco
+	`)
+)
+
+type RegenerateMCOOptions struct {
+	RESTClientGetter genericclioptions.RESTClientGetter
+
+	mcoNamespace   string
+	modifyUserData bool
+	outputNewCert  bool
+
+	genericclioptions.IOStreams
+}
+
+func NewCmdRegenerateTopLevel(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *cobra.Command {
+	o := &RegenerateMCOOptions{
+		RESTClientGetter: restClientGetter,
+		IOStreams:        streams,
+	}
+
+	cmd := &cobra.Command{
+		Use:                   "regenerate-mco",
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Regenerate the machine config operator certificates in an OpenShift cluster"),
+		Long:                  regenerateMCOLong,
+		Example:               regenerateMCOExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(run(o, args))
```

Comment:
typically, `Run()` is a method of the options, that get `Validate()`-ed before use